tests/forms_tests/field_tests/test_jsonfield.py

```diff
@@ -97,6 +97,20 @@ class JSONForm(Form):
         form = JSONForm({'json_field': '["bar"]'}, initial={'json_field': ['foo']})
         self.assertIn('[&quot;foo&quot;]</textarea>', form.as_p())
 
+    def test_formfield_value_none(self):
```

Comment:
I'm not sure about these tests. I was expecting to see something more like this 🤔

https://github.com/django/django/blob/64839512a6ed04a29e49e246acf8337b1be2cb8e/tests/forms_tests/field_tests/test_charfield.py#L28

Reply:
`clean()` works without this patch. The issue is in rendering a bound field. We should check `errors` and `as_p()`.

Reply:
```suggestion
    def test_redisplay_none_input(self):
```

Reply:
Yes, sorry. I was wondering why the tests are looking at rendering which is concern of widgets/forms rather than the behaviour of the field itself. 

I guess the behaviour of the field informs what the widget/form render?

Reply:
> I guess the behaviour of the field informs what the widget/form render?

Yes.